lib/ferto/client.rb

```diff
@@ -46,22 +49,29 @@ def initialize(opts = {})
     #
     # @example
     #   downloader = Ferto::Client.new
-    #   downloader.download(
+    #   dl_resp = downloader.download(
     #     aggr_id: 'msystems',
     #     aggr_limit: 3,
     #     url: 'http://foo.bar/a.jpg',
-    #     callback_url: 'http://example.com/downloads/myfile',
+    #     callback_type: 'http',
+    #     callback_dst: 'http://example.com/downloads/myfile',
     #     extra: { groupno: 'foobar' }
     #   )
     #
-    # @return [Ferto::Client::Response]
+    # @raise [Ferto::ConnectionError] if the client failed to connect to the
+    #   downloader API
+    #
+    # @return [Ferto::Response]
     def download(aggr_id:, aggr_limit: @aggr_limit, url:,
```

Comment:
IMHO we should document the arguments of this method. Yes, they are documented in the server's README already, but this is not quite the same.

For example, the driver accepts a hash for the `extra` argument but the server expects a string. Also, in the future we could use a `URI` object for the `callback_url`, while the server expects a String.

Reply:
That said, this is out-of-scope of this PR and could be done in another step.

Reply:
Agreed. I would suggest that you open an issue to keep it noted and leave it outside the scope of this PR.

Reply:
I don't think it deserves an issue so I'll just push this to master.

Reply:
https://github.com/skroutz/ferto

Reply:
https://github.com/skroutz/ferto/pull/6